App.tsx

```diff
@@ -17,57 +17,82 @@ interface Product {
   discountedPrice: number;
   description: string;
 }
- 
+
 function App(): JSX.Element {
-const [page , setPage] = useState(1);
-const [products, setProducts] = useState<Product[]>([]);
-const [loading, setLoading] = useState(true);
+  const [page, setPage] = useState(1);
+  const [products, setProducts] = useState<Product[]>([]);
+  const [loading, setLoading] = useState(true);
 
-const apiData = async()=>{
-  try{
-    const res = await fetch(`${'https://dummyjson.com/products'}?limit=05&page=${page}`)
```

Comment:
Write a Custom hook to encapsulate all the buisness logic inside a custom hook like api calling, pagination etc.

Reply:
Done